python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2447,7 +2490,11 @@ def handle_new_event(
                 and event.get_dagster_event().is_job_event
             ):
                 self._run_storage.handle_run_event(run_id, event.get_dagster_event())
-
+                run = self.get_run_by_id(run_id)
+                if run and event.get_dagster_event().event_type == DagsterEventType.RUN_FAILURE:
+                    self.add_run_tags(
```

Comment:
is there a potential for an issue here if we can't access the DB? like if the run fails because of a DB connection issue, we won't be able to write this tag and then the run will never get retried. Not sure if this is really something to worry about though

Reply:
note that in cloud, handles_run_events_in_store_event is True, so we will need to make sure that this change is propagated to cloud as well (which helps with the case that you mentioned as well)

Reply:
for future traceability - see https://github.com/dagster-io/internal/pull/12765